compaction/compaction_manager.cc

```diff
@@ -348,6 +348,16 @@ future<compaction_manager::compaction_stats_opt> compaction_manager::perform_tas
 }
 
 future<> compaction_manager::on_compaction_completion(table_state& t, sstables::compaction_completion_desc desc, sstables::offstrategy offstrategy) {
+    auto& cs = get_compaction_state(&t);
+    auto new_sstables = boost::copy_range<std::unordered_set<sstables::shared_sstable>>(desc.new_sstables);
+    for (const auto& sst : desc.old_sstables) {
+        if (!new_sstables.contains(sst)) {
+            cs.sstables_requiring_cleanup.erase(sst);
+        }
+    }
+    if (cs.sstables_requiring_cleanup.empty()) {
+        cs.owned_ranges_ptr = nullptr;
+    }
     return t.on_compaction_completion(std::move(desc), offstrategy);
```

Comment:
Do we still need this same code in compacting_sstable_registration::release_compacting() ?

Reply:
Actually I think we need it for incremental compaction.

Reply:
Without this, the compacted sstables aren't removed incrementally.